llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -47384,6 +47408,24 @@ static SDValue combineShiftLeft(SDNode *N, SelectionDAG &DAG) {
   ConstantSDNode *N1C = dyn_cast<ConstantSDNode>(N1);
   EVT VT = N0.getValueType();
 
+  // Exploits AVX2 VSHLV/VSRLV instructions for efficient unsigned vector shifts
+  // with out-of-bounds clamping.
+  if (N0.getOpcode() == ISD::VSELECT &&
+      supportedVectorVarShift(VT, DAG.getSubtarget<X86Subtarget>(), ISD::SHL)) {
```

Comment:
Sorry I misunderstood you're question - just add the `const X86Subtarget &Subtarget` arg to combineShiftLeft to match combineShiftRightArithmetic

Reply:
done